Expose GodotString formatting facility

lufterc opened the pull request:
This PR exposes `godot_string_format` API method to make built-in Godot string formatting available in GDNative API. It also adds a basic formatting in-engine test case.

Bromeon commented:
Thanks!

With the suggested API, we don't support the `placeholder` argument, but I think that one is used so rarely that it should be OK. We can always add an "overload" later.

Could you maybe add an `assert_eq!` in the test that makes sure that the old string is unchanged? Just to make sure, and also for expressing the expected behavior.

lufterc commented:
> Do you think it makes sense to document the method with 2 examples, one for array and one for dictionary? Maybe also add a quick array test.

Sure, documentation always may come in handy :) I've added examples for index-style array values and dictionary, and also added test assertions for original string and array.

Bromeon commented:
Combined the commits and changed ` ```ignore` to ` ```no_run`, so we can run compile-time doc tests on the examples. Retained author information with signed-off-by.

Thanks again!
bors r+
